adapters/kalao-avalanche/index.js

```diff
@@ -0,0 +1,145 @@
+            },
+        );
+        return resp.data;
+    };
+
+    getPrice = async timestamp => {
+        const resp = await axios.get(
+            `${URL}/token-price?key=${KEY}&token_address=${this.token}&protocol=${this.protocol}&timestamp=${timestamp}`,
+            {
+                headers: {
+                    "User-Agent":
+                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4619.141 Safari/537.36",
+                },
+            },
+        );
+        return resp.data;
+    };
+
+    stop = async () => {
+        this.sdk.stop();
+    };
+
+    getBuyerInfo = async transaction => {
+        const pubrchaseLog = transaction.logs.find(
+            l => l.topics[0] === "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef", // Purchase Event Log
+        );
+        if (!pubrchaseLog) {
+            return;
+        }
+        const nftContract = pubrchaseLog.address;
+        const buyer = "0x" + pubrchaseLog.topics[2].substr(-40, 40);
+        const tokenId = parseInt(pubrchaseLog.topics[3], 16);
+
+        return {
+            nftContract,
+            tokenId,
+            buyer,
+        };
+    };
+
+    getSeller = async (dealId, block) => {
+        const seller = await this.sdk.callContractMethod("getSeller", [dealId], [undefined, block.number]);
+        return seller;
+    };
+
+    _getPrice = async (dealId, block) => {
+        if (!this.symbol.decimals) {
+            return { price: null, priceUsd: null };
+        }
+
+        const price = await this.sdk.callContractMethod("getPrice", [dealId], [undefined, block.number]);
+        const po = await this.getPrice(block.timestamp);
+        const nativePrice = new BigNumber(price).dividedBy(10 ** this.symbol.decimals);
+
+        return {
+            price: nativePrice.toNumber(),
+            priceUsd: nativePrice.multipliedBy(po.price).toNumber(),
+        };
+    };
+
+    process = async event => {
+        const dealId = parseInt(event.data.substr(0, 66), 16);
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+        const transaction = await this.sdk.getTransactionReceipt(event.transactionHash);
+        const buyerInfo = await this.getBuyerInfo(transaction);
+        if (!buyerInfo) {
+            return;
+        }
+        const seller = await this.getSeller(dealId, block);
+        if (!seller) {
+            return;
+        }
+        const { price, priceUsd } = await this._getPrice(dealId, block);
+
+        const entity = {
+            provider_name: this.name,
+            provider_contract: this.contract,
+            protocol: this.protocol,
+            nft_contract: buyerInfo.nftContract,
+            nft_id: buyerInfo.tokenId,
+            token: this.token,
+            token_symbol: this.symbol.symbol,
+            amount: 1,
+            price,
+            price_usd: priceUsd,
```

Comment:
`token_symbol` is undefined
`price` and `price_usd` are null
Make sure to pass the tests before requesting a review